swap_meet/vendor.py

```diff
@@ -1,2 +1,109 @@
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        # Remove `my_item` from this Vendor's inventory and add it to the other Vendor's inventory
+        self.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        # Remove `their_item` from the other Vendor's inventory and add it to this Vendor's inventory
+        other_vendor.inventory.remove(their_item)
+        self.add(their_item)
+        # If successful, return True
+        return True
+    
+    # def swap_first_item(self, other_vendor):
+    #     # Check if either the vendor or the friend have empty inventory, and return False if so
+    #     if not self.inventory or not other_vendor.inventory:
+    #         return False
+    #     # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+    #     return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+    def swap_first_item(self, other_vendor):
+        try:
+            # Check if either the vendor or the friend have empty inventory, and return False if so
+            if not self.inventory or not other_vendor.inventory:
+                return False
+            # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        except IndexError:
+            return False
+
+    def get_by_category(self, category):
+        # Create an empty list to store the matching items
+        items_matching_category = []
+        # Iterate through the inventory to find items with matching category
+        for item in self.inventory:
+            if item.get_category() == category:
+                items_matching_category.append(item)
+        # Return the list of matching items
+        return items_matching_category
+
+
+    def get_best_by_category(self, category):
+        # Check if there are any items in the inventory with the given category
+        if not self.get_by_category(category):
+            # If there are no items in the inventory with the given category, return None
+            return None
+        # If there are items in the inventory with the given category, find the one with the highest condition
+        return max(self.get_by_category(category), key=lambda item: item.condition)
+
+
+    
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        # Check if the current vendor has an item of the category specified by their_priority
+        # or if the other vendor has an item of the category specified by my_priority.
+        if not self.get_best_by_category(their_priority) or not other_vendor.get_best_by_category(my_priority):
+            # If either of these conditions is not met, return False as the swap cannot happen.
+            return False
```

Comment:
You can eliminate this guard clause and just `return self.swap_items(other_vendor, my_item, their_item)` below, since `swap_items` handles `None` values.